BilanCarbone/src/Pages/ListeUtilisateur.jsx

```diff
@@ -149,9 +149,11 @@ function ListeUtilisateur() {
         const fetchRoles = async () => {
         const rolesData = {};
           for (const utilisateur of utilisateurs) {
-            rolesData[utilisateur.id] = await getRoleUtilisateur(utilisateur.id);
+            rolesData[utilisateur.userRepresentation.id] = await getRoleUtilisateur(utilisateur.userRepresentation.id);
+            
```

Comment:
essaie d'utiliser Promise.all a la place de la boucle for pour faire le fetch du role de chaque utilisateurs